aten/src/ATen/native/Cross.cpp

```diff
@@ -42,9 +43,41 @@ Tensor & cross_out(const Tensor & input, const Tensor & other, const c10::option
     TORCH_CHECK(input.size(dim) == 3, "dimension ", dimension.value(), " does not have size 3");
   }
 
-  if (out.sizes() != input.sizes()) {
-    out.resize_as_(input);
-  }
+  // check if resizing output is required
+  // raise a warning while resizing if output has one or more elements
+  at::native::resize_output(out, input.sizes());
+
+  cross_stub(device1, out, input, other, dim);
+  return out;
+}
```

Comment:
Given that `torch.cross` is just `torch.linalg.cross` with a funny default for `dim`, you could just write `cross_out` as performing the funny search if `dim` is `None` and pass this value to `at::native::linalg_cross_out`, which would then perform all the checks. This would avoid quite a bit of repeated code.
In this case, you could also mark `cross` as `CompositeImplicitAutograd` and you could delete its formulae for the gradients.

Reply:
This was not that trivial, involved a bit more refactoring but I think it is concise and crisp now.
`cross` now completely dispatches through `linalg_cross`.